### PR TITLE
Notifications for metre when a task is complete. 

### PR DESCRIPTION
**Pull request by georgethomas111**
@gsquire @johnhof

### CODE REVIEW

**Comment by gsquire** [comment]
Ignoring whitespace in the diff: https://github.com/gospackler/metre/pull/5/files?w=1

File: messages.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	Message     string  `json:"message"`
+}
+
+func parseMessage(msg string) (*trackMessage, error) {
+	trackMsg := new(trackMessage)
+	err := json.Unmarshal([]byte(msg), trackMsg)
+	return trackMsg, err
+}
+
+func createMsg(mt msgType, id string, uid string, msg string) string {
```
Comment (gsquire) [inline messages.go]: You can pass arguments of the same type like this:

``` go
func createMsg(mt msgType, id, uid, msg string) string {}
```

File: metre.go [hunk 2]
```diff
@@ -2,118 +2,177 @@
...
+	if cErr != nil {
+		return Metre{}, cErr
+	}
+	q, qErr := NewQueue(queueUri)
+	if qErr != nil {
+		return Metre{}, qErr
+	}
+
+	t, tErr := NewQueue(trackQueueUri)
+	if qErr != nil {
```
Comment (gsquire) [inline metre.go]: Should this be `tErr`?

File: metre.go [hunk 3]
```diff
@@ -2,118 +2,177 @@
...
+	m.TaskMap[id] = t
+	t.MessageChannel = m.MessageChannel
+	m.Cron.AddFunc(t.Interval, func() {
+		m.scheduleFromId(id)
+	})
+}
+
+func (m *Metre) scheduleFromId(ID string) (string, error) {
+	t, ok := m.TaskMap[ID]
+	if ok == false {
```
Comment (gsquire) [inline metre.go]: This can be simplified to:

``` go
if t, ok := m.TaskMap[ID]; !ok {}
```

File: metre.go [hunk 4]
```diff
@@ -2,118 +2,177 @@
...
-
-    tr := NewTaskRecord(t.ID)
-    t.Process(tr, m.Scheduler, m.Cache, m.Queue)
-    return buildTaskKey(tr), nil
+	t, ok := m.TaskMap[ID]
+	if ok == false {
+		return "", errors.New("task [" + ID + "] not recognized")
+	}
+
+	tr := NewTaskRecord(t.GetID())
```
Comment (gsquire) [inline metre.go]: Why the switch to the `GetID` function?

File: metre_test.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
+package metre
+
+import (
+	"testing"
+
+	"fmt"
+	log "github.com/Sirupsen/logrus"
```
Comment (gsquire) [inline metre_test.go]: Sort this import.

File: metre_test.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		return
+	},
+	Process: func(t TaskRecord, s Scheduler, c Cache, q Queue) {
+		log.Info("Processing Test  " + t.UID)
+		return
+	},
+}
+
+func TestMain(m *testing.M) {
+	m.Run()
```
Comment (gsquire) [inline metre_test.go]: I generally use `os.Exit(m.Run())`.

File: metre.go (hunk 2 above)
Comment (georgethomas111) [inline metre.go]: Oops yeah ! evil of copy pasting.

File: task.go [hunk 7]
```diff
@@ -1,8 +1,65 @@
...
+
+func (t Task) GetID() string {
+	return t.ID
+}
+
+func (t Task) GetInterval() string {
+	return t.Interval
+}
+
+func (t *Task) checkComplete() bool {
```
Comment (gsquire) [inline task.go:30]: This can be shortened to `return t.MessageCount == t.ScheduleCount && t.ScheduleDone`

File: metre.go (hunk 4 above)
Comment (georgethomas111) [inline metre.go]: Not needed.

**Pull request merged**